lib/_tls_wrap.js

```diff
@@ -1091,6 +1091,15 @@ exports.connect = function(/* [port, host], options, cb */) {
       socket._hadError = true;
       var error = new Error('socket hang up');
       error.code = 'ECONNRESET';
+      if (!options.socket) {
+        if (options.path && !options.port) {
+          error.path = options.path;
+        } else {
+          error.port = options.port;
+          error.host = options.host;
+          error.localAddress = options.localAddress;
+        }
```

Comment:
Why not directly setting:

```js
error.path = options.path;
error.port = options.port;
error.host = options.host;
error.localAddress = options.localAddress
```

The user just needs information, then we provide them in faster way without the extra checking.